recipes/nifty/meta.yaml

```
@@ -0,0 +1,68 @@
+{% set version = "v1.0.8" %}
+{% set checksum = "4a70eb106a926271cf86cae3e568c71b26ca40ecdfbfc337110f485410aa2876" %}
+
+package:
+  name: nifty
+  version: {{ version }}
+
+source:
+    url: https://github.com/DerThorsten/nifty/archive/{{ version }}.tar.gz
+    sha256: {{ checksum }}
+
+build:
+  number: 0
+  skip: True  # [win and vc<14]
+  skip: True  # [py2k]
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - ninja  # [win]
+  host:
+    - python
+    - numpy
+    - xtensor >=0.21,<0.22
+    - xtensor-python >=0.24,<0.25
+    - vigra
+    - hdf5 >=1.10  # [not win]
+    - z5py >=2.0.8  # [not win]
+    # build dependencies needed for z5
+    - nlohmann_json  # [not win]
+    - blosc  # [not win]
+    - bzip2  # [not win]
+    - zlib  # [not win]
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
+    - hdf5 >=1.10  # [not win]
+    - scikit-image
+    - blosc  # [not win]
+    - bzip2  # [not win]
+    - zlib  # [not win]
+    # macos needs boost as rt dependency because of boost filesystem
+    - {{ pin_compatible('boost-cpp') }}  # [osx]
```

The global pinning will take care of the compatibility (this would be overly loose, only pinning on major version)
```suggestion
    - boost-cpp  # [osx]
```